### DISCUSSION TITLE
scrolling text

### DISCUSSION THREAD

**Discussion by BKSalman** [hecrj/iced#1413] [Q&A] [answered]
I checked the examples of the crate, tour had a text input that was adding it's value to a text element, I'm interested in how does the text element scroll when the text overflows, because it doesn't behave like that when I do it, it just wraps to a new line

desired behavior 
![image](https://user-images.githubusercontent.com/85521119/184994185-ef800358-e555-4d95-a771-8ac112206971.png)

what I get 
![image](https://user-images.githubusercontent.com/85521119/184994368-6f508960-9dc1-48d2-b020-bad987591d6d.png)

my view 
```
    fn view(&mut self) -> Element<'_, Self::Message> {
        container::Container::new(
                Text::new(&self.key)
                .width(iced::Length::Fill)
                .horizontal_alignment(alignment::Horizontal::Center)
                .height(iced::Length::Fill)
            )
        .into()
    }

```

**Comment by tarkah** [comment, marked as answer]
Text wraps by default. The only way to get it to "scroll" is by using the text input. If you just want it to be cut off on a single row of text, you can make the following change:

```rs
    fn view(&mut self) -> Element<'_, Self::Message> {
        container::Container::new(
            Text::new(&self.key).height(iced::Length::Units(20))
        )
        .width(iced::Length::Fill)
        .height(iced::Length::Fill)
        .center_x()
        .into()
    }
```

Iced uses default font size of 20. Match the height of the `Text` widget with the font size so it doesn't wrap.

**Reply by BKSalman** [reply]
thank you for replying!

is it possible to get the "scrolling" effect by making a custom widget?
